cmake/developer_package/python_requirements.cmake

```diff
@@ -97,8 +97,46 @@ function(ov_check_pip_packages)
 
     if(PYTHONINTERP_FOUND)
         execute_process(
-            COMMAND ${PYTHON_EXECUTABLE} -c "import pkg_resources ;
-            pkg_resources.require('onnx==1.12.0')"
+            COMMAND ${PYTHON_EXECUTABLE} -c "
+import pkg_resources ;
+import re ;
+import os ;
+constraints = {} ;
+constraints_path = None ;
+requirements = [] ;
+with open('${ARG_REQUIREMENTS_FILE}') as f:
+    raw_requirements = f.readlines() ;
+for line in raw_requirements:
+    if line.startswith('-c'):
+        constraints_path = os.path.join(os.path.dirname('${ARG_REQUIREMENTS_FILE}'), line.split(' ')[1][:-1]) ;
+if constraints_path: 
+    with open(constraints_path) as f:
+        raw_constraints = f.readlines() ;
+    for line in raw_constraints:
+        if line.startswith('#') or line=='\\n':
+            continue ;
+        line = line.replace('\\n', '') ;
+        package, delimiter, constraint = re.split('(~|=|<|>|;)', line, maxsplit=1) ;
+        if constraints.get(package) is None:
+            constraints[package] = [delimiter + constraint] ;
+        else:
+            constraints[package].extend([delimiter + constraint]) ;
+    for line in raw_requirements:
+        if line.startswith(('#', '-c')):
+            continue ;
+        line = line.replace('\\n', '') ;
+        if re.search('\\W', line):
+            requirements.append(line) ;
+        else:
+            constraint = constraints.get(line) ;
+            if constraint:
+                for marker in constraint: 
+                    requirements.append(line+marker) ;
+            else:
+                requirements.append(line) ;
+else:
+    requirements = raw_requirements ;
+pkg_resources.require(requirements)"
```

Comment:
we can put it into python script inside this folder and call inside the `execute_process`

Reply:
I tried it and there were issues due to cmake being called from different directories, which results in the Python script not being found due to relative paths. I'll get back to it once the CI gets green.

Reply:
we need to use path relative to folder with IEDevScripts.cmake
Example https://github.com/openvinotoolkit/openvino/blob/73bedced873d75d509f06d237e70ba7cfbb52208/cmake/developer_package/api_validator/api_validator.cmake#L169

Reply:
Thank you, that helped. I moved the script into a separate `.py` file.